File: deliveryRoute/src/main/java/com/sparta/deliveryroute/application/exception/CompanyException.java
package com.sparta.deliveryroute.application.exception;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.experimental.FieldDefaults;

@Getter
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
public class CompanyException extends RuntimeException {

	Error error;

	public CompanyException(Error error) {
		this.error = error;
	}
}

File: company/src/main/kotlin/com/sparta/company/application/exception/CompanyException.kt
package com.sparta.company.application.exception

open class CompanyException(val error: Error) : RuntimeException()

class NotFoundCompanyException : CompanyException(Error.NOT_FOUND_COMPANY)

class IncorrectCompanyTypeException : CompanyException(Error.INCORRECT_COMPANY_TYPE)

class IncorrectHubIdException : CompanyException(Error.INCORRECT_HUB_ID)

class AccessDeniedException : CompanyException(Error.ACCESS_DENIED)